src/csync/csync_update.cpp

```diff
@@ -387,18 +387,13 @@ static int _csync_detect_update(CSYNC *ctx, std::unique_ptr<csync_file_stat_t> f
   CSYNC_LOG(CSYNC_LOG_PRIORITY_INFO, "file: %s, instruction: %s <<=", fs->path.constData(),
       csync_instruction_str(fs->instruction));
 
+  QByteArray path = fs->path;
   switch (ctx->current) {
     case LOCAL_REPLICA:
-      if (c_rbtree_insert(ctx->local.tree, (void *) fs.release()) < 0) {
-        ctx->status_code = CSYNC_STATUS_TREE_ERROR;
-        return -1;
-      }
+      ctx->local.files.emplace(path, std::move(fs));
```

Comment:
s/emplace/insert/

Reply:
It would need to be `insert(std::make_pair(path, std::move(fs))`, is this better? I'm changing the vector to push_back, but emplace would make sense here I think (although can't really understand why would anybody use insert unless they already have the key/value pair).

Reply:
Done